include/itkMorphologicalContourInterpolator.hxx

```diff
@@ -0,0 +1,193 @@
+            if (val != 0 || (m_Label!=0 && val==m_Label))
+            {
+                typename TImage::RegionType bb1;
+                bb1.SetIndex(ind);
+                for (int a = 0; a < TImage::ImageDimension; a++)
+                    bb1.SetSize(a, 1);
+                std::pair<BoundingBoxesType::iterator, bool> resBB
+                    = m_BoundingBoxes.insert(std::make_pair(val, bb1));
+                if (!resBB.second) //include this index in existing BB
+                    ExpandRegion(resBB.first->second, ind);
+
+                std::pair<OrientationsType::iterator, bool> res
+                    = m_Orientations.insert(std::make_pair(val, ors));
+                OrientationsType::iterator oRef = res.first;
+                int cTrue = 0;
+                for (int a = 0; a < TImage::ImageDimension; a++)
+                {
+                    indPrev = ind;
+                    indPrev[a]--;
+                    indNext = ind;
+                    indNext[a]++;
+                    {
+                        if (input->GetPixel(indPrev) == 0 && input->GetPixel(indNext) == 0)
+                        {
+                            ors[a] = true;
+                            cTrue++;
+                        }
+                    }
+                }
+                if (cTrue == 1) //slice has empty adjacent space only along one axis
+                {
+                    for (int a = 0; a < TImage::ImageDimension; a++)
+                        oRef->second[a] = oRef->second[a] || ors[a]; //add this dimension for this label
+                }
+            }
+            ++it;
+        }
+
+        //widen bounding boxes by 1
+        region = output->GetRequestedRegion();
+        for (BoundingBoxesType::iterator it = m_BoundingBoxes.begin(); it != m_BoundingBoxes.end(); ++it)
+        {
+            for (int a = 0; a < TImage::ImageDimension; a++)
+            {
+                if (it->second.GetIndex(a) -1 <= region.GetIndex(a))
+                    it->second.SetIndex(a, it->second.GetIndex(a) - 1);
+                if (it->second.GetIndex(a) + it->second.GetSize(a) + 1 
+                    <= region.GetIndex(a) + region.GetSize(a))
+                    it->second.SetSize(a, it->second.GetSize(a) + 1);
+            }
+        }
+    }
+
+    template< class TImage>
+    void MorphologicalContourInterpolator<TImage>::
+        InterpolateAlong(int axis, typename TImage::Pointer out)
+    {
+
+    }
+
+    template< class TImage>
+    void MorphologicalContourInterpolator<TImage>::GenerateData()
+    {
+        typename TImage::ConstPointer input = this->GetInput();
+        typename TImage::Pointer output = this->GetOutput();
+        this->AllocateOutputs();
+
+        if (m_Axis == -1)
+        {
+            this->DetermineSliceOrientations();
+
+            OrientationType aggregate = OrientationType();
+            if (this->m_Label == 0)
+            {
```

Comment:
Please fix indentation to Kitware/ITK style throughout (two spaces indent after if)